File: Lista/Java/Q13.java
class Matematica {

    public static long fatorial(int numero) {
        if (numero < 0) {
            throw new IllegalArgumentException("O número deve ser não-negativo.");
        }
        long resultado = 1;
        for (int i = 1; i <= numero; i++) {
            resultado *= i;
        }
        return resultado;
    }
}

public class Main {
    public static void main(String[] args) {
        int numero = 5;
        long resultado = Matematica.fatorial(numero);
        System.out.println("Fatorial de " + numero + " é: " + resultado);

        numero = 10;
        resultado = Matematica.fatorial(numero);
        System.out.println("Fatorial de " + numero + " é: " + resultado);

        try {
            numero = -3;
            resultado = Matematica.fatorial(numero);
            System.out.println("Fatorial de " + numero + " é: " + resultado);
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: Lista/Python/Q13.py
class Matematica:
    @staticmethod
    def fatorial(n):
        if n < 0:
            raise ValueError("O fatorial não está definido para números negativos.")
        elif n == 0 or n == 1:
            return 1
        else:
            resultado = 1
            for i in range(2, n + 1):
                resultado *= i
            return resultado

numero = 5
print(f'O fatorial de {numero} é: {Matematica.fatorial(numero)}')

numero_negativo = -3
try:
    print(f'O fatorial de {numero_negativo} é: {Matematica.fatorial(numero_negativo)}')
except ValueError as e:
    print(e)